Makefile

```diff
@@ -4,65 +4,114 @@ TARGETS ?= unexpected.js unexpected.js.map
+
+build/externaltests: externaltests/*
+	@babel --copy-files --out-dir build/externaltests --quiet externaltests
+
+build: build/lib build/test build/externaltests
+
 .PHONY: ${TARGETS}
-${TARGETS}: lib/*
-	./node_modules/.bin/rollup --config --sourcemap --format umd --name weknowhow.expect -o unexpected.js lib/index.js
+${TARGETS}: build
+	./node_modules/.bin/rollup --config --sourcemap --format umd --name weknowhow.expect -o unexpected.js build/lib/index.js
 
-create-html-runners: test/tests.tpl.html test/JasmineRunner.tpl.html
+create-html-runners: build/test test/tests.tpl.html test/JasmineRunner.tpl.html
 	@for file in tests JasmineRunner ; do \
-		(sed '/test files/q' ./test/$${file}.tpl.html | sed '$$d' && \
+		(sed '/test files/q' ./build/test/$${file}.tpl.html | sed '$$d' && \
 		find test -name '*.spec.js' | sed 's/test/    <script src="./' | sed 's/$$/"><\/script>/' && \
-		sed -n '/test files/,$$p' ./test/$${file}.tpl.html | sed '1d') > ./test/$${file}.html;\
+		sed -n '/test files/,$$p' ./build/test/$${file}.tpl.html | sed '1d') > ./build/test/$${file}.html;\
 	done
 
 .PHONY: create-html-runners
 
 test-phantomjs: create-html-runners ${TARGETS}
-	phantomjs ./node_modules/mocha-phantomjs-core/mocha-phantomjs-core.js test/tests.html spec "`node -pe 'JSON.stringify({useColors:true,grep:process.env.grep})'`"
-
-test-jest-if-supported-node-version:
-ifeq ($(shell node --version | grep -vP '^v[0123]\.'),)
-	@echo Skipping, jest is unsupported with node $(shell node --version)
-else
-	./node_modules/.bin/jest
-endif
+	phantomjs ./node_modules/mocha-phantomjs-core/mocha-phantomjs-core.js build/test/tests.html spec "`node -pe 'JSON.stringify({useColors:true,grep:process.env.grep})'`"
 
 test-jasmine:
 	./node_modules/.bin/jasmine JASMINE_CONFIG_PATH=test/support/jasmine.json
 
 test-jasmine-browser: create-html-runners ${TARGETS}
+	@echo open http://localhost:5000/build/test/JasmineRunner.html
 	@./node_modules/.bin/serve .
 
+test-sources:
+ifeq ($(NODE_VERSION), $(shell cat .nvmrc))
+MOCHA_OPTS = ./test/mocha.opts
+TEST_SOURCES = $(shell find test -name '*.spec.js')
+else
+	make build
+MOCHA_OPTS = ./build/test/mocha.opts
+TEST_SOURCES = $(shell find build/test -name '*.spec.js')
+endif
+
+.PHONY: test-jest
+test-jest:
+ifeq ($(NODE_VERSION), $(shell cat .nvmrc))
+	./node_modules/.bin/jest
+else
+	./node_modules/.bin/jest -c test/jest.es5.config.json
+endif
+
+test-jest-if-supported-node-version:
+ifeq ($(shell node -p -e '/^v[0123]\./.test(process.version)'),true)
+	@echo Skipping, jest is unsupported with node $(shell node --version)
+else
+	make test-jest
+endif
+
 .PHONY: test
-TEST_SOURCES = $(shell find test -name '*.spec.js') $(shell find documentation -name '*.md')
-test: lint
-	@./node_modules/.bin/mocha $(TEST_SOURCES)
+test: test-sources
+	@./node_modules/.bin/mocha --opts $(MOCHA_OPTS) $(TEST_SOURCES) $(TEST_SOURCE_MARKDOWN)
+
+nyc-includes:
+ifeq ($(NODE_VERSION), $(shell cat .nvmrc))
```

Comment:
Same here

Reply:
I took the assumption that you would use `nvm use` when developing. But you are right that it would be an improvement to use semver to check that you are on a larger version than required by `.nvmrc`.

Reply:
I guess we can use https://www.npmjs.com/package/assert-node-version for the check.